### PR TITLE
Improve model-actions e2e tests

### LINKED ISSUE #32683: Fix Flaky model-actions tests
3 biggest flakes right now:

Screen Shot 2023-07-26 at 3 41 50 PM

### PR DESCRIPTION
**Pull request by iethree**
closes https://github.com/metabase/metabase/issues/32683

### Description

Our top three flakiest e2e specs are all from this group

Screen Shot 2023-07-26 at 3 41 50 PM

I was able to reproduce these flakes locally, though not very consistently.

All of the flakes seem to follow the same pattern: they almost never fail completely, but most of the time, the first run (or 2 or 3) fails. My hypothesis is that there are some code paths that need to heat up, and are slow on the first try before they get working. Unclear if this is at the js, clojure, database, or disk level. possibly a combination of all of them.

This mostly adds some appropriate waits throughout to make sure relevant API requests have completed before moving on.

### CODE REVIEW

File: e2e/test/scenarios/actions/model-actions.cy.spec.js [hunk 1]
```diff
@@ -591,10 +603,16 @@ describe(
...
-      cy.findByRole("button", { name: /Create basic actions/i }).click();
+      createBasicActions();
 
       openActionEditorFor("Create");
 
+      cy.wait("@getAction").then(({ response }) => {
+        const { parameters, visualization_settings } = response.body;
+        expect(parameters).to.have.length(5);
+        expect(visualization_settings).to.have.property("fields");
+      });
```
Comment (iethree) [inline e2e/test/scenarios/actions/model-actions.cy.spec.js:614]: This test was failing because the `GET` action api request was return an action without parameters or visualization_settings, so there was no form to render at all. I'm not sure this wait actually helps, but I couldn't reproduce the failure again after I added it. In any event, these assertions themselves should fail and make the cause of the failure clearer than just "form didn't show up"

fail | pass
--- | ---
Screen Shot 2023-07-26 at 3 06 41 PM | Screen Shot 2023-07-26 at 3 07 27 PM

File: e2e/test/scenarios/actions/model-actions.cy.spec.js [hunk 2]
```diff
@@ -847,6 +867,7 @@ function enableSharingFor(actionName, { publicUrlAlias }) {
   cy.findByRole("dialog").within(() => {
     cy.button("Action settings").click();
     cy.findByLabelText("Make public").should("not.be.checked").click();
+    cy.wait("@enableActionSharing");
```
Comment (iethree) [inline e2e/test/scenarios/actions/model-actions.cy.spec.js:870]: I think some failures were because sharing hadn't been properly enabled because we weren't ever waiting for the api request to complete before navigating to a new page.

File: e2e/test/scenarios/actions/model-actions.cy.spec.js [hunk 3]
```diff
@@ -714,6 +732,8 @@ describe(
 
       cy.findByRole("button", { name: "Update" }).click();
 
+      cy.wait("@updateAction");
```
Comment (iethree) [inline e2e/test/scenarios/actions/model-actions.cy.spec.js:735]: This is a classic cause of many flakes. We click a button in the UI that fires off an API request, then we expect to do something with application state that depends on that API request.

In many cases, we're just looking for data that isn't there yet. In this case, we navigate to a new page, and I think sometimes cancel some API requests, so they never even complete.

**TLDR; Always wait on API requests that you're triggering.**

**Review by ranquild — APPROVED** [review]
